src/types/mutation.ts

```diff
@@ -72,7 +73,7 @@ export type Mutation<
   TAction extends MutationAction,
   TPlaceholder extends unknown,
 > = PromiseLike<AwaitedMutation<TAction>> & {
-  data: MutationData<TAction> | TPlaceholder,
+  data: MutationData<TAction> | DefaultValue<TPlaceholder, undefined>,
```

Comment:
Found this bug in the earlier mutation code. Without this change the data was always `unknown` because the placeholder was always `unknown`